src/kagglehub/clients.py

```diff
@@ -317,6 +318,10 @@ def __init__(self) -> None:
             "X-KAGGLE-PROXY-DATA": data_proxy_token,
         }
 
+        iap_token = os.getenv(KAGGLE_IAP_TOKEN_ENV_VAR_NAME)
+        if iap_token:
+            self.headers["Authorization"] = f"Bearer {iap_token}"
```

Comment:
If this header is already used for Kaggle API token, you can use the `Proxy-Authorization` header. I use something like this on staging for testing:

```
!curl -H "Proxy-Authorization: Bearer $KAGGLE_IAP_TOKEN" -H "Authorization: Bearer $(cat $KAGGLE_API_V1_TOKEN)" https://staging.kaggle.com/api/v1/hello
```

Reply:
`X-Kaggle-Authorization` is used for the JWT.
`X-KAGGLE-PROXY-DATA` is used for the data proxy token.

`Authorization` is currently not used so the IAP token should be sent through that header to the webtier JWT handler.

A quick test showed it still failed but needed more time to investigate but I switched to something else instead.

Given that you are here, if the IAP token is sent through the `Authorization` header, should it work? 

My next step was to look at the logic in the webtier to find where the IAP token is read.

Reply:
If the `Authorization` header isn't used already, it should work. GCP accepts `Proxy-Authorization` in case `Authorization` is used in app's auth mechanism but that doesn't seem to be the case here since we use a custom header: `X-Kaggle-Authorization`. There is a class called `IapValidator` in the webtier that validates this token but I don't know much about it

Reply:
I see, we don't go through the IAP proxy though.

The request is sent to the data proxy (allows `kagglehub.model_download(...)/dataset_download(...)` to work when internet is disabled).

The data proxy then forwards the request to the webtier via the Kubernetes internal DNS: https://github.com/Kaggle/kaggleazure/blob/fea4c58f37a662a71721d5b1434897c80b84c4c5/Kaggle.Emulators.DataProxy/Controllers/KaggleJwtHandlerController.cs#L23

This way, if you talk to the staging data-proxy, then the staging webtier is called.

I will stop working on this for now given it is low priority because it only affects admin who haven't joined the competition (see: https://b.corp.google.com/issues/381445306#comment13).

Reply:
Okay, I see. That's a completely separate path then. I don't have a concrete suggestion :)